packages/web-components/fast-components/src/accordion/accordion-item/accordion-item.template.ts

```diff
@@ -0,0 +1,45 @@
+import { html, ref } from "@microsoft/fast-element";
+import { endTemplate, startTemplate } from "../../patterns/start-end";
+import { AccordionItem } from "./accordion-item";
+
+export const AccordionItemTemplate = html<AccordionItem>`
+    <template
+        class="${x => (x.expanded ? "expanded" : "")}"
+        slot="item"
+    >
+        <div
+            class="heading"
+            part="heading"
+            role="heading"
+            aria-level="${x => x.headinglevel}"
+        >
+            <button
+                class="button"
+                part="button"
+                ${ref("button")}
+                aria-expanded="${x => x.expanded}"
+                aria-controls="${x => x.id}-panel"
+                id="${x => x.id}"
+                @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+            >
+                <span class="heading">
+                    <slot name="heading" part="heading"></slot>
+                </span>
+            </button>
+            ${startTemplate}
+            ${endTemplate}
+            <span class="glyph" part="glyph">
+                <slot name="expanded-glyph" part="expanded-glyph"></slot>
+                <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
```

Comment:
Clever! Assuming this is toggled based on state. Do we think this is preferable as an implementor to dynamically changing/passing a different icon based on state?

Reply:
its hidden and shown via css based on `:host(.expanded)`. Some cases require different Icons + or - for example some may just rotate a chevron > this allows for both since everything can be managed via css.

Reply:
sure - I understand the implementation here. My primary ask is whether we think this is preferred from an API standpoint vs. a single slot that changes based on state. For context, this is somewhat driving the question: https://github.com/WICG/open-ui/pull/54#discussion_r411660381

We don't need to land as part of this, but we should keep an eye on how things like these land in Open-UI and align where we can.

Reply:
Curious how would a single slot work like the above, how would an author pass different icons?

Reply:
If I'm using SVG's I could likely conditionally render based on the expanded state of the component. If I want to use a CSS solution I can do that as well with that approach. If I want to use icon fonts I can swap a class name based on a single span sent into the slot.

Again, not prescribing that approach, just something to think about and keep an eye on (discussion above). I think this is a clever approach regardless of how that discussion pans out :)

Reply:
cool I'll resolve this for now then.